components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/event/queue/EventListMapManager.java

```diff
@@ -0,0 +1,200 @@
+                long lastSequenceIdForApp = -1;
+
+                if (perAppLastControlMessageSequenceNumberList.size() != 0) {
+                    lastSequenceIdForApp = perAppLastControlMessageSequenceNumberList.get(siddhiAppName);
+                }
+                synchronized (this) {
+                    //we need this block synchronized to ensure if last remembered siddhi app id is smaller than the
+                    // event id then add it to the tree map synchronously, else older event may add to the tree map
+                    if (sequenceID > lastSequenceIdForApp) {
+                        String attributes;
+                        int attributeLength = eventContent.getInt();
+                        if (attributeLength == 0) {
+                            throw new InvalidByteMessageException("Invalid attributeLength size = 0");
+                        } else {
+                            attributes = BinaryMessageConverterUtil.getString(eventContent, attributeLength);
+                        }
+                        String[] attributeTypes = attributes.substring(1, attributes.length() - 1).split(", ");
+                        events[i] = SiddhiEventConverter.getEvent(eventContent, attributeTypes);
+                        queuedEvent = new QueuedEvent(siddhiAppName, sourceHandlerElementId, sequenceID, events[i]);
+                        this.addToEventListMap(sequenceID, queuedEvent);
+                    }
+
+                    if (log.isDebugEnabled()) {
+                        if (startTime == 0L) {
+                            startTime = new Date().getTime();
+                        }
+                        count++;
+                        if (count % TPS_EVENT_THRESHOLD == 0) {
+                            endTime = new Date().getTime();
+                            log.info("# of events batch : " + count + "start timestamp : " + startTime +
+                                    " end time stamp : " + endTime + " Throughput is (events / sec) : " +
+                                    (((TPS_EVENT_THRESHOLD * 1000) / (endTime - startTime))) +
+                                    " Total Event Count : " + count);
+                            startTime = new Date().getTime();
+                        }
+                    }
+                }
+
+
+            }
+        } catch (UnsupportedEncodingException e) {
+            log.error("Error when converting bytes " + e.getMessage(), e);
+        }
+    }
+
+    public void trimAndSendToInputHandler() throws InterruptedException {
+        Map<String, SiddhiAppData> siddhiAppMap = StreamProcessorDataHolder.getStreamProcessorService().
+                getSiddhiAppMap();
+
+        for(Map.Entry<Long,QueuedEvent> listMapValue : eventListMap.entrySet()) {
+            long key = listMapValue.getKey();
+            QueuedEvent queuedEvent = listMapValue.getValue();
+            for (Map.Entry<String, SiddhiAppData> entry : siddhiAppMap.entrySet()) {
```

Comment:
Instead of looping through all siddhi apps can we do `siddhiAppMap.get(queuedEvent.getSiddhiAppName());` and do a null check?

Reply:
+1